src/libaktualizr/package_manager/packagemanagerinterface.cc

```diff
@@ -67,6 +72,15 @@ static int ProgressHandler(void* clientp, curl_off_t dltotal, curl_off_t dlnow,
   if (ds->progress_cb && progress > ds->last_progress) {
     ds->last_progress = progress;
     ds->progress_cb(ds->target, "Downloading", progress);
+    //OTA-4864:Improve binary file download progress logging. Report each XX sec report event that notify user
+    if (!ds->target.IsOstree()) {      
+      auto now = std::chrono::steady_clock::now();
+      auto milliseconds = std::chrono::duration_cast<std::chrono::milliseconds>(now - ds->time_lastreport);
+      if(milliseconds.count() > LogProgressInterval || progress == 100) {
```

Comment:
I don't think you need to do the `progress == 100` check, because once the download is complete, a separate event is sent to make that clear, so that's already covered.

Reply:
done